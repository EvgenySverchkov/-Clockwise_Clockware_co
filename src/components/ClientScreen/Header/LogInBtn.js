import React from "react";
import { Link } from "react-router-dom";

function LogIn() {
  return (
    <li className="nav-item">
      <Link className="nav-link" to="/client/login">
        Login
      </Link>
    </li>
  );
}

export default LogIn;
