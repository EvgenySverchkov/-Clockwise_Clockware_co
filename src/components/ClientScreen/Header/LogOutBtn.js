import React from "react";
import { Link } from "react-router-dom";

function LogOutBtn({ handler }) {
  return (
    <li className="nav-item">
      <Link className="nav-link" to="/client" onClick={handler}>
        Logout
      </Link>
    </li>
  );
}

export default LogOutBtn;
