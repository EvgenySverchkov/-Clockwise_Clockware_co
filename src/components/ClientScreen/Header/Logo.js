import React from "react";
import {Link} from "react-router-dom";
import PropTypes from "prop-types";

const Logo = ({children}) => (
    <Link to="/client" className="navbar-brand">
          {children}
    </Link>
);

Logo.propTypes = {
    children: PropTypes.node.isRequired
}

export default Logo;