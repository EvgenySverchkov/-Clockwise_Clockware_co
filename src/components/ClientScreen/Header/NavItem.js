import React from "react";
import { Link } from "react-router-dom";

import PropTypes from "prop-types";

const NavItem = ({title, link}) => (
    <li className="nav-item">
        <Link className="nav-link" to={link}>
            {title}
        </Link>
    </li>
);

NavItem.propTypes = {
    title: PropTypes.string,
    link: PropTypes.string
}

export default NavItem;