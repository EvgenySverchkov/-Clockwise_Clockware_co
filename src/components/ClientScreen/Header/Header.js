import React from "react";

import Logo from "./Logo";
import NavList from "./NavBtnsList";

function Header() {
  return (
    <header className="masthead mb-auto">
      <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
        <Logo>Clockwise Co.</Logo>
        <NavList />
      </nav>
    </header>
  );
}

export default Header;
