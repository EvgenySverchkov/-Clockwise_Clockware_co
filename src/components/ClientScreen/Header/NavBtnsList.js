import React from 'react';
import { connect } from "react-redux";
import PropTypes from "prop-types";

import LogOutBtn from "./LogOutBtn";
import LogInBtn from "./LogInBtn";
import NavItem from "./NavItem";

const NavList = ({isAuth}) => (
    <div className="navbar-collapse collapse" id="navbarsExample03">
        <ul className="navbar-nav mr-auto">
            {isAuth ? <LogOutBtn /> : <LogInBtn />}
            <NavItem title = {"SignUp"} link={"/client/registration"}/>
            <NavItem title = {"To main"} link={"/client"}/>
        </ul>
    </div>
);

NavList.propTypes = {
    isAuth: PropTypes.bool
}

function mapStateToProps(state) {
    return {
      isAuth: state.client_order_reduser.isAuth,
    };
}

export default connect(mapStateToProps)(NavList);