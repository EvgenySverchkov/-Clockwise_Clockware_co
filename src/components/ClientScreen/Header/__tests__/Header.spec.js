import React from "react";
import {create} from "react-test-renderer";
import {MemoryRouter} from "react-router-dom";
import {Provider} from "react-redux";
import configMockStore from "redux-mock-store";

import Header from "../Header";

const mockStore = configMockStore();


describe("Test of <Header /> on client side", ()=>{
    it("If client is unlogged", ()=>{
        const store = mockStore({
            authReducer: {
                isAuthClient: false
            }
        })
        const component = create(
            <Provider store = {store}>
                <MemoryRouter initialEntries={["/"]}>
                    <Header />
                </MemoryRouter>
            </Provider>
    
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
    
    it("If client is logged", ()=>{
        const store = mockStore({
            authReducer: {
                isAuthClient: true
            }
        })
        const component = create(
            <Provider store = {store}>
                <MemoryRouter initialEntries={["/"]}>
                    <Header />
                </MemoryRouter>
            </Provider>
    
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});