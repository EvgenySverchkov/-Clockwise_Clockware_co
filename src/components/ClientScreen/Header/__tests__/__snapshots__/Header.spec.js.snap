// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <Header /> on client side If client is logged 1`] = `
<header
  className="masthead mb-auto"
>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/client"
      onClick={[Function]}
    >
      Clockwise Co.
    </a>
    <button
      aria-controls="navbarsExample03"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      data-target="#navbarsExample03"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="navbar-collapse collapse"
      id="navbarsExample03"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/client"
            onClick={[Function]}
          >
            To main
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/client/userProfile"
            onClick={[Function]}
          >
            Profile
          </a>
        </li>
      </ul>
    </div>
  </nav>
</header>
`;

exports[`Test of <Header /> on client side If client is unlogged 1`] = `
<header
  className="masthead mb-auto"
>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/client"
      onClick={[Function]}
    >
      Clockwise Co.
    </a>
    <button
      aria-controls="navbarsExample03"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      data-target="#navbarsExample03"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="navbar-collapse collapse"
      id="navbarsExample03"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/client/login"
            onClick={[Function]}
          >
            Login
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/client/registration"
            onClick={[Function]}
          >
            SignUp
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/client"
            onClick={[Function]}
          >
            To main
          </a>
        </li>
      </ul>
    </div>
  </nav>
</header>
`;
